internal/pubsub/pubsub.go

```diff
@@ -84,7 +90,10 @@ func (t *Topic) Send(ctx context.Context, m *Message) error {
 // It only returns after all pending messages have been sent.
 func (t *Topic) Close() error {
 	t.batcher.Flush()
-	return t.driver.Close()
+	t.mu.Lock()
```

Comment:
Shouldn't you do this before the `Flush`? otherwise you have a race where you `Flush` but then new `Send` calls come in before you lock+mark closed.

Reply:
Done.